ddtrace/internal/datadog/profiling/dd_wrapper/include/sample.hpp

```diff
@@ -0,0 +1,91 @@
+#pragma once
+
+#include "libdatadog_helpers.hpp"
+#include "profile.hpp"
+#include "types.hpp"
+
+#include <array>
+#include <string>
+#include <string_view>
+#include <vector>
+
+extern "C"
+{
+#include "datadog/profiling.h"
+}
+
+namespace Datadog {
+
+class SampleManager; // friend
+
+class Sample
+{
+  private:
+    static inline Profile profile_state{};
```

Comment:
Should this be a pointer to the global state?

Reply:
Added as a TODO for the testing phase